.github/workflows/ci_workflows.yml

```diff
@@ -126,60 +126,16 @@ jobs:
       test_command: pytest --pyargs erfa
       targets: |
         # Linux wheels
-        - cp39-manylinux_x86_64
-        - cp310-manylinux_x86_64
-        - cp311-manylinux_x86_64
-        - cp39-musllinux_x86_64
-        - cp310-musllinux_x86_64
-        - cp311-musllinux_x86_64
-        - cp39-manylinux_aarch64
-        - cp310-manylinux_aarch64
-        - cp311-manylinux_aarch64
+        - cp3*-manylinux_x86_64
+        - cp3*-musllinux_x86_64
+        - cp3*-manylinux_aarch64
         # MacOS X wheels - we deliberately do not build universal2 wheels.
         # Note that the arm64 wheels are not actually tested so we
         # rely on local manual testing of these to make sure they are ok.
-        - cp39*macosx_x86_64
-        - cp310*macosx_x86_64
-        - cp311*macosx_x86_64
-        - cp39*macosx_arm64
-        - cp310*macosx_arm64
-        - cp311*macosx_arm64
+        - cp3*macosx_x86_64
+        - cp3*macosx_arm64
         # Windows wheels
-        - cp39*win32
-        - cp310*win32
-        - cp311*win32
-        - cp39*win_amd64
-        - cp310*win_amd64
-        - cp311*win_amd64
-    secrets:
-      pypi_token: ${{ secrets.pypi_token }}
-
-  # TEMPORARY WORKAROUND: Have to separate this out to use unreleased NumPy for Python 3.12
-  # because shell command to conditionally set PIP_PRE is broken in OpenAstronomy/github-actions-workflows
-  # https://github.com/OpenAstronomy/github-actions-workflows/issues/152
-  build_and_publish_py312:
-
-    uses: OpenAstronomy/github-actions-workflows/.github/workflows/publish.yml@v1
-    with:
-      upload_to_pypi: ${{ startsWith(github.ref, 'refs/tags/v') && !endsWith(github.ref, '.dev') }}
-      test_extras: test
-      test_command: pytest --pyargs erfa
-      targets: |
-        # Linux wheels
-        - cp312-manylinux_x86_64
-        - cp312-musllinux_x86_64
-        - cp312-manylinux_aarch64
-        # MacOS X wheels - we deliberately do not build universal2 wheels.
-        # Note that the arm64 wheels are not actually tested so we
-        # rely on local manual testing of these to make sure they are ok.
-        - cp312*macosx_x86_64
-        - cp312*macosx_arm64
-        # No NumPy wheel for win32
-        #- cp312*win32
-        - cp312*win_amd64
-      # Required so that cp312 can grab a pre-release numpy.
-      # Can remove when cp312 is released.
-      env: |
-        CIBW_ENVIRONMENT: PIP_PRE=1
```

Comment:
I don't grok the internals of erfa, so I cannot review but I wonder why you remove the extra job for py312. I split it out on purpose because py312 needs pre-release of numpy.

Reply:
See https://github.com/liberfa/pyerfa/pull/94#issuecomment-1710775593 - with the limited API we don't build Python 3.12 wheels anymore. We just build 3.9 wheels and test them on 3.10+. Testing them on 3.12 is disabled for now because we can't have Python 3.12 as a separate job anymore with this PR.

If you would be more comfortable waiting for a Python 3.12 release of Numpy we can do that, but it's not strictly needed - if we know via other CI config that Python 3.12 works fine then testing the wheels on 3.9, 3.10 and 3.11 is probably sufficient.

Reply:
Hmm... What about using artifact upload/download for the py312? Though maybe numpy will ship py312 wheel anyway when 1.26 is released, so maybe we can just wait it out. Thanks for the clarification!

Reply:
Happy to just wait for 1.26 final, no point overcomplicating things.